src/Optimizely/Notification/NotificationCenter.php

```diff
@@ -0,0 +1,151 @@
+                return -1;
+            }
+        }
+
+        $this->_notifications[$notification_type][$this->_notificationId] = $notification_callback;
+        $this->logger->log(Logger::INFO, "Callback added for notification type '{$notification_type}'.");
+        $returnVal = $this->_notificationId++;
+        return $returnVal;
+    }
+
+    /**
+     * [removeNotificationListener description]
+     * @param  [type] $notification_id [description]
+     * @return [type]                  [description]
+     */
+    public function removeNotificationListener($notification_id)
+    {
+        foreach ($this->_notifications as $notification_type => $notifications) {
+            foreach (array_keys($notifications) as $id) {
+                if ($notification_id == $id) {
+                    unset($this->_notifications[$notification_type][$id]);
+                    $this->logger->log(Logger::INFO, "Callback with notification ID '{$notification_id}' has been removed.");
+                    return true;
+                }
+            }
+        }
+
+        $this->logger->log(Logger::DEBUG, "No Callback found with notification ID '{$notification_id}'.");
+        return false;
+    }
+
+    /**
+     * [clearNotifications description]
+     * @param  [type] $notification_type [description]
+     * @return [type]                    [description]
+     */
+    public function clearNotifications($notification_type)
+    {
+        if (!NotificationType::isNotificationTypeValid($notification_type)) {
+            $this->logger->log(Logger::ERROR, "Invalid notification type.");
+            return null;
+        }
+
+        $this->_notifications[$notification_type] = [];
+        $this->logger->log(Logger::INFO, "All callbacks for notification type '{$notification_type}' have been removed.");
+    }
+
+    /**
+     * [clearAllNotifications description]
+     * @return [type] [description]
+     */
+    public function clearAllNotifications()
+    {
+        foreach(array_values(NotificationType::getAll()) as $type){
+            $this->_notifications[$type] = [];
+        }
+    }
+
+    /**
+     * [fireNotifications description]
+     * @param  [type] $notification_type [description]
+     * @param  array  $args              [description]
+     * @return [type]                    [description]
+     */
+    public function fireNotifications($notification_type, array $args = [])
+    {
+        if (!isset($this->_notifications[$notification_type])) {
+            return;
+        }
+
+        foreach (array_values($this->_notifications[$notification_type]) as $callback) {
+            try {
+                call_user_func_array($callback, $args);
+            } catch (Exception $e) {
+                $this->logger->log(Logger::ERROR, "Problem calling notify callback.");
+            } catch (ArgumentCountError $e) {
+                $this->logger->log(Logger::ERROR, "Problem calling notify callback.");
```

Comment:
can u plz. change to be more specific, what actually the issue was. 
ArgumentCountError must come first than Exception.

Reply:
Yes, I will work on this today

Reply:
btw, there is no option here it can go into 2nd exception, either ArgumentCountError or Exception. means u won't have 100% coverage of this method. Remove one exception.